x86/efi/run

```diff
@@ -30,32 +30,20 @@ fi
 shift 1
 
 # Prepare EFI boot file system
-#   - Copy .efi file to host dir $EFI_TEST/$EFI_CASE/
+#   - Copy .efi file to host dir $EFI_TEST/$EFI_CASE/EFI/BOOT/BOOTX64.EFI
 #     This host dir will be loaded by QEMU as a FAT32 image
-#   - Make UEFI startup script that runs the .efi on boot
-mkdir -p "$EFI_TEST/$EFI_CASE/"
-if [ $EFI_CASE != "_NO_FILE_4Uhere_" ]; then
-  cp "$EFI_SRC/$EFI_CASE.efi" "$EFI_TEST/$EFI_CASE/"
-else
-  touch "$EFI_TEST/$EFI_CASE/$EFI_CASE.efi"
```

Comment:
We cannot do this same trick here with this new approach to get the test cases with bad QEMU args to be marked as `SKIP`? Let me try.

Reply:
Ah.., I see you're doing it further below... But in the pull request, you mentioned it's not working for the `vmexit` test case? Odd... Let me try it out.

Reply:
Actually, this patch is not working at all on my setup :-(. Any idea what's going on? If not, I can debug.

upstreaming-uefi-v2-gh1:
$ ./run_tests.sh
PASS apic-split (48 tests)
PASS ioapic-split (19 tests)
...

upstreaming-uefi-v2-gh2:
$ ./run_tests.sh
FAIL apic-split (timeout; duration=90s)
FAIL ioapic-split (timeout; duration=90s)

Reply:
Emm it works on my workstation, have you tried to `make distclean` then configure - make - run_tests?

I posted a testing result in the main pull-request discussion thread.

Reply:
Brand new repo... I'll dig into it and report back with more details on what's going wrong then.

Reply:
Maybe the UEFI OVMF you are using does not recognize this default filename? How about running a single test using `./x86/efi/run`, does it work?

Reply:
Ack. I'll be offline for the afternoon. I'll try to follow back up later tonight. Sorry for the slowness on my side! Note, I'm not opposed to sending this upstream and I can reply there as well. I don't want to hold you up. But if you don't mind waiting a little longer, I'll follow up as soon as I can (tonight or tomorrow). Thanks!

Reply:
Ah don't worry, I'll hold this patchset for a few more days so we can discuss more details if possible. Thank you for your time!

Reply:
Here's what I see when I run msr.efi. I'll start debugging. But I agree, maybe my UEFI is not recognizing the default filename.

```
$ ./x86/efi/run x86/msr.efi
/usr/bin/qemu-system-x86_64 --no-reboot -nodefaults -device pc-testdev -device isa-debug-exit,iobase=0xf4,iosize=0x4 -vnc none -serial stdio -device pci-testdev -machine accel=kvm -drive file=/usr/share/ovmf/OVMF.fd,format=raw,if=pflash,readonly=on -drive file.dir=efi-tests/msr/,file.driver=vvfat,file.rw=on,format=ra
w,if=virtio -net none -nographic -m 256 -smp 1
BdsDxe: loading Boot0002 "EFI Internal Shell" from Fv(7CB8BDC9-F8EB-4F34-AAEA-3EE4AF6516A1)/FvFile(7C04A583-9E3E-4F1C-AD65-E05268D0B4D1)
BdsDxe: starting Boot0002 "EFI Internal Shell" from Fv(7CB8BDC9-F8EB-4F34-AAEA-3EE4AF6516A1)/FvFile(7C04A583-9E3E-4F1C-AD65-E05268D0B4D1)
UEFI Interactive Shell v2.2
EDK II
UEFI v2.70 (EDK II, 0x00010000)
Mapping table
      FS0: Alias(s):HD0b:;BLK1:
          PciRoot(0x0)/Pci(0x3,0x0)/HD(1,MBR,0xBE1AFDFA,0x3F,0xFBFC1)
     BLK0: Alias(s):
          PciRoot(0x0)/Pci(0x3,0x0)
Press ESC in 1 seconds to skip startup.nsh or any other key to continue.
Shell>


^C
```

Reply:
> Maybe the UEFI OVMF you are using does not recognize this default filename? How about running a single test using ./x86/efi/run, does it work?

After poking around a little, I think this is a good theory. How can I debug this? Maybe we can do a video call if you have time?

Reply:
This is what I see when I run `./x86/efi/run ./x86/msr.efi`, which skips the `EFI Interactive Shell` output

```
BdsDxe: loading Boot0001 "UEFI Misc Device" from PciRoot(0x0)/Pci(0x3,0x0)
BdsDxe: starting Boot0001 "UEFI Misc Device" from PciRoot(0x0)/Pci(0x3,0x0)
enabling apic
PASS: MSR_IA32_SYSENTER_CS
PASS: MSR_IA32_SYSENTER_ESP
PASS: MSR_IA32_SYSENTER_EIP
PASS: MSR_IA32_MISC_ENABLE
PASS: MSR_IA32_CR_PAT
PASS: MSR_FS_BASE
PASS: Expected #GP on WRSMR(MSR_FS_BASE, 0xaaaaaaaaaaaaaaaa), got vector 13
PASS: MSR_GS_BASE
PASS: Expected #GP on WRSMR(MSR_GS_BASE, 0xaaaaaaaaaaaaaaaa), got vector 13
PASS: MSR_KERNEL_GS_BASE
PASS: Expected #GP on WRSMR(MSR_KERNEL_GS_BASE, 0xaaaaaaaaaaaaaaaa), got vector 13
PASS: MSR_EFER
PASS: MSR_LSTAR
PASS: Expected #GP on WRSMR(MSR_LSTAR, 0xaaaaaaaaaaaaaaaa), got vector 13
PASS: MSR_CSTAR
PASS: Expected #GP on WRSMR(MSR_CSTAR, 0xaaaaaaaaaaaaaaaa), got vector 13
PASS: MSR_SYSCALL_MASK
SUMMARY: 17 tests
```

When I change the `x86/efi/run` to not copying any EFI binary to `efi-tests/`, my UEFI OVMF Shell also shows:

```
❯ ./x86/efi/run ./x86/msr.efi
/usr/bin/qemu-system-x86_64 --no-reboot -nodefaults -device pc-testdev -device isa-debug-exit,iobase=0xf4,iosize=0x4 -vnc none -serial stdio -device pci-testdev -machine accel=kvm -drive file=/usr/share/edk2-ovmf/x64/OVMF_CODE.fd,format=raw,if=pflash,readonly=on -drive file.dir=efi-tests/msr/,file.driver=vvfat,file.rw=on,format=raw,if=virtio -net none -nographic -m 256 -smp 1

BdsDxe: failed to load Boot0001 "UEFI Misc Device" from PciRoot(0x0)/Pci(0x3,0x0): Not Found
BdsDxe: loading Boot0002 "EFI Internal Shell" from Fv(7CB8BDC9-F8EB-4F34-AAEA-3EE4AF6516A1)/FvFile(7C04A583-9E3E-4F1C-AD65-E05268D0B4D1)
BdsDxe: starting Boot0002 "EFI Internal Shell" from Fv(7CB8BDC9-F8EB-4F34-AAEA-3EE4AF6516A1)/FvFile(7C04A583-9E3E-4F1C-AD65-E05268D0B4D1)
UEFI Interactive Shell v2.2
EDK II
UEFI v2.70 (EDK II, 0x00010000)
Mapping table
      FS0: Alias(s):HD0b:;BLK1:
          PciRoot(0x0)/Pci(0x3,0x0)/HD(1,MBR,0xBE1AFDFA,0x3F,0xFBFC1)
     BLK0: Alias(s):
          PciRoot(0x0)/Pci(0x3,0x0)
Press ESC in 1 seconds to skip startup.nsh or any other key to continue.
Shell>
```

Reply:
> 
> 
> > Maybe the UEFI OVMF you are using does not recognize this default filename? How about running a single test using ./x86/efi/run, does it work?
> 
> After poking around a little, I think this is a good theory. How can I debug this? Maybe we can do a video call if you have time?

Sure, I have time right now, maybe send me a GVC so I can join?